eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Utxo.hs

```diff
@@ -65,5 +251,273 @@ allegraToConwayUtxoPredFailure = \case
+
+  {-   adaID ∉ supp mint tx - check not needed because mint field of type MultiAsset
+   cannot contain ada -}
+
+  {-   ∀ txout ∈ allOuts txb, getValue txout ≥ inject (serSize txout ∗ coinsPerUTxOByte pp) -}
+  let allSizedOutputs = txBody ^. allSizedOutputsTxBodyF
+  runTest $ Babbage.validateOutputTooSmallUTxO pp allSizedOutputs
+
+  let allOutputs = fmap sizedValue allSizedOutputs
+  {-   ∀ txout ∈ allOuts txb, serSize (getValue txout) ≤ maxValSize pp   -}
+  runTest $ Alonzo.validateOutputTooBigUTxO pp allOutputs
+
+  {- ∀ ( _ ↦ (a,_)) ∈ allOuts txb,  a ∈ Addrbootstrap → bootstrapAttrsSize a ≤ 64 -}
+  runTestOnSignal $ Shelley.validateOutputBootAddrAttrsTooBig allOutputs
+
+  netId <- liftSTS $ asks networkId
+
+  {- ∀(_ → (a, _)) ∈ allOuts txb, netId a = NetworkId -}
+  runTestOnSignal $ Shelley.validateWrongNetwork netId allOutputs
+
+  {- ∀(a → ) ∈ txwdrls txb, netId a = NetworkId -}
+  runTestOnSignal $ Shelley.validateWrongNetworkWithdrawal netId txBody
+
+  {- (txnetworkid txb = NetworkId) ∨ (txnetworkid txb = ◇) -}
+  runTestOnSignal $ Alonzo.validateWrongNetworkInTxBody netId txBody
+
+  {- txsize tx ≤ maxTxSize pp -}
+  runTestOnSignal $ Shelley.validateMaxTxSizeUTxO pp tx
+
+  {-   totExunits tx ≤ maxTxExUnits pp    -}
+  runTest $ Alonzo.validateExUnitsTooBigUTxO pp tx
+
+  {-   ‖collateral tx‖  ≤  maxCollInputs pp   -}
+  runTest $ Alonzo.validateTooManyCollateralInputs pp txBody
+
+  trans @(EraRule "UTXOS" era) =<< coerce <$> judgmentContext
+
+--------------------------------------------------------------------------------
+-- ConwayUTXO STS
+--------------------------------------------------------------------------------
+
+instance
+  forall era.
+  ( EraTx era
+  , EraUTxO era
+  , ConwayEraTxBody era
+  , AlonzoEraTxWits era
+  , Tx era ~ AlonzoTx era
+  , EraRule "UTXO" era ~ ConwayUTXO era
+  , InjectRuleFailure "UTXO" ShelleyUtxoPredFailure era
+  , InjectRuleFailure "UTXO" Allegra.AllegraUtxoPredFailure era
+  , InjectRuleFailure "UTXO" AlonzoUtxoPredFailure era
+  , InjectRuleFailure "UTXO" BabbageUtxoPredFailure era
+  , InjectRuleFailure "UTXO" ConwayUtxoPredFailure era
+  , -- instructions for calling UTXOS from BabbageUTXO
+    Embed (EraRule "UTXOS" era) (ConwayUTXO era)
+  , Environment (EraRule "UTXOS" era) ~ UtxoEnv era
+  , State (EraRule "UTXOS" era) ~ UTxOState era
+  , Signal (EraRule "UTXOS" era) ~ Tx era
+  , PredicateFailure (EraRule "UTXO" era) ~ ConwayUtxoPredFailure era
+  ) =>
+  STS (ConwayUTXO era)
+  where
+  type State (ConwayUTXO era) = UTxOState era
+  type Signal (ConwayUTXO era) = AlonzoTx era
+  type Environment (ConwayUTXO era) = UtxoEnv era
+  type BaseM (ConwayUTXO era) = ShelleyBase
+  type PredicateFailure (ConwayUTXO era) = ConwayUtxoPredFailure era
+  type Event (ConwayUTXO era) = AlonzoUtxoEvent era
+
+  initialRules = []
+
+  -- TODO: It would be great to be able to use the Alonzo `utxoTransition` here
+  -- i.e: transitionRules = [Alonzo.utxoTransition @era]
```

Comment:
Actually, the `UTXOW` transitions allow this because unlike in `utxoTransition`s, `utxowTransition`s doesn't concretise `EraRule "UTXO" era`. These only get concretised for the instances in `UTXOW`s case which allows the reuse of the transition functions. Would it be possible or okay to modify the `utxoTransition` in `Alonzo` and remove `EraRule "UTXO" era ~ AlonzoUTXO era` so I can reuse that in `Conway`s `utxoTransition`?

Reply:
You should not be defining `utxoTransition` for `Conway` nor the `ConwayUTXO` rule type. It can all be reused from Babbage. That is the problem here.

Reply:
I was trying to reuse the rule initially but that lead to (for all the `InjectRuleFailure` instances): 
```    
• Couldn't match type: ConwayUtxoPredFailure (ConwayEra c)
                 with: BabbageUtxoPredFailure (ConwayEra c)`
```
Which makes sense, since we are reusing the rule from `Babbage` and in the `STS (BabbageUTXO)` instance we specify the `PredicateFailure` type to be `BabbageUtxoPredFailure era` (line 464 in `Babbage` `Utxo.hs`). This is what prevents me from reusing the rule when I also want to use `ConwayUtxoPredFailure`. That's why I thought that I'll have to do something similar to the `STS (ConwayUTXOW)` instance, which is using `babbageUtxowTransition` for `transitionRules`. However, I also can't reuse `utxoTransition` from `Babbage` (without modifying it) because it has the `EraRule "UTXO" era ~ BabbageUTXO era` constraint.

Is there a way to reuse the `UTXO` rules from `Babbage` while also using `ConwayUtxoPredFailure` for predicate failures?

Reply:
> Is there a way to reuse the UTXO rules from Babbage while also using ConwayUtxoPredFailure for predicate failures?

Yes there is and that is exactly the goal.

Reply:
Actually, my bad, I do see the problem that you are running into. So, you do need `ConwayUTXO` type and a new `STS` instance, like you did in this PR, because you are setting the new predicate failure. But you can reuse the `utxoTransition` from here: https://github.com/IntersectMBO/cardano-ledger/blob/2c9ba10d085331fa99840a63b40b8e8b716fba5d/eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxo.hs#L343-L361

The problem is in the fact that the type signature makes this restriction: `EraRule "UTXO" era ~ BabbageUTXO era`, which we can get rid of by:
* removing that type equality constraint
* replacing `BabbageUTXO` with `EraRule "UTXO" era` in `Embed (EraRule "UTXOS" era) (BabbageUTXO era)` 
* and adding a few mores constrains (which ones I can't tell without building the code).

Let me know if it doesn't unblock you and we can talk more about it.

Reply:
>  Would it be possible or okay to modify the utxoTransition in Alonzo and remove EraRule "UTXO" era ~ AlonzoUTXO era so I can reuse that in Conways utxoTransition?

Important note on the above question. In Conway we are reusing UTXO rule from Babbage, not Alonzo